### ISSUE TITLE
[Usage]: gpt_attention does not support slide window causal mask

### ISSUE THREAD

**Issue by 0Unicorn** [NVIDIA/TensorRT-LLM#10620]
### System Info

GPU:NVIDIA L20
tensorrt_llm:1.2.0

### How would you like to use TensorRT-LLM

The parameter description for `gpt_attention` includes 
`* tensorrt_llm.layers.AttentionMaskType.sliding_window_causal for GPT`

https://github.com/NVIDIA/TensorRT-LLM/blob/2967d299fb05cb66af50b0792de0bcb80fb4f4df/tensorrt_llm/functional.py#L5459-L5467

However, in actual use, an assertion prevents the use of sliding_window.
https://github.com/NVIDIA/TensorRT-LLM/blob/2967d299fb05cb66af50b0792de0bcb80fb4f4df/tensorrt_llm/layers/attention.py#L1081-L1088

Does `gpt_attention` currently support `sliding window`, or is it only possible to use `plain TensorRT mode`?

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and checked the [documentation](https://nvidia.github.io/TensorRT-LLM/) and [examples](https://github.com/NVIDIA/TensorRT-LLM/tree/main/examples) for answers to frequently asked questions.

**Comment by karljang** [comment]
@0Unicorn ,

Sliding window attention IS supported in TensorRT-LLM with the GPT attention plugin, but NOT through AttentionMaskType.sliding_window_causal.

##### Correct usage:
1. Set attention_mask_type=AttentionMaskType.causal
2. Configure sliding_window in your model config
3. The FMHA kernels and KV cache manager handle the sliding window behavior automatically

##### Example models using this approach:
  - Gemma3 (with sliding_window config parameter)
  - Mistral models with sliding window attention
